pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -597,6 +608,233 @@ def test_optimize(self):
+        """Test failure sync advisories.
+
+        If update_dates and update_version are the same, pkglist intersection is non-empty
+        and not equal to either pkglist sync should fail.
+
+        Tested error_msg must be same as we use in pulp_rpm.app.advisory.
+        """
+        body = gen_rpm_remote(RPM_UNSIGNED_FIXTURE_URL)
+        remote = self.remote_api.create(body)
+
+        # sync
+        repo, remote = self.do_test(remote=remote)
+
+        # create remote with colliding advisory
+        body = gen_rpm_remote(RPM_ADVISORY_INCOMPLETE_PKG_LIST_URL)
+        remote = self.remote_api.create(body)
+
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        task_result = monitor_task(sync_response.task)
+        error_msg = 'Incoming and existing advisories have the same id ' \
+            'and timestamp but different and intersecting package lists. ' \
+            'At least one of them is wrong. Advisory id: {}'.format(RPM_ADVISORY_TEST_ID)
+
+        # add resources to clean up
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        self.assertIn(error_msg, task_result['error']['description'])
+
+    @unittest.skip(
+        'FIXME: Enable this test after https://pulp.plan.io/issues/6605 is fixed'
+    )
+    def test_sync_advisory_no_updated_date(self):
+        """Test sync advisory with no update.
+
+        1. Sync repository with advisory which has updated_date
+        2. Re-sync with repo with same id and version as previous
+           but missing updated_date (issued_date should be used instead).
+        """
+        body = gen_rpm_remote(RPM_UNSIGNED_FIXTURE_URL)
+        remote = self.remote_api.create(body)
+
+        # sync
+        repo, remote = self.do_test(remote=remote)
+
+        body = gen_rpm_remote(RPM_ADVISORY_NO_DATES)
+        remote = self.remote_api.create(body)
+
+        # re-sync
+        repo, remote = self.do_test(repo, remote)
+
+        # add resources to clean up
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        added_advisory_date = [
+            advisory['updated_date']
+            for advisory in get_added_content(repo.to_dict())[PULP_TYPE_ADVISORY]
+            if RPM_ADVISORY_TEST_ID in advisory['id']
+        ]
+        removed_advisory_date = [
+            advisory['issued_date']
+            for advisory in get_removed_content(repo.to_dict())[PULP_TYPE_ADVISORY]
+            if RPM_ADVISORY_TEST_ID in advisory['id']
+        ]
+
+        self.assertGreater(
+            parse_datetime(added_advisory_date[0]),
+            parse_datetime(removed_advisory_date[0])
+        )
+
+    def test_sync_advisory_updated_update_date(self):
+        """Test sync advisory with no update or issue date."""
```

Comment:
Is it correct doc string? I thought the goal here is not to test the absence of any date but to test that the newest one is added to a repo version.